numpy/core/arrayprint.py

```diff
@@ -743,20 +755,22 @@ def __init__(self, precision, sign=False):
 
     def __call__(self, x):
         if isnan(x):
-            if self.sign:
-                return '+' + _nan_str
+            nan_str = _format_options['nanstr']
+            if self.sign == '+':
+                return '+' + nan_str
             else:
-                return ' ' + _nan_str
+                return ' ' + nan_str
```

Comment:
Again not your change, but this is not the same as for regular float (where no space is added)

Reply:
Yes we discussed this. I haven't updated LongFloat because it needs a total rewrite. (see the 3rd column in the big set of changed tests below).